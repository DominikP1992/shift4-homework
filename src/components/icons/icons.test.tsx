import { render } from '@testing-library/react';
import * as Icons from './index';

describe('Icon ', () => {
  Object.entries(Icons).forEach(([IconName, IconComponent]) => {
    it(`${IconName} renders without crashing`, () => {
      const { container } = render(<IconComponent />);
      expect(container.firstChild).toBeInTheDocument();
    });

    it(`${IconName} matches snapshot`, () => {
      const { container } = render(<IconComponent />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
