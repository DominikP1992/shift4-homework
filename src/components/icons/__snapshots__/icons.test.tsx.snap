// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon  ChevronLeft matches snapshot 1`] = `
<svg
  fill="none"
  height="12"
  width="8"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M6 12 0 6l6-6 1.4 1.4L2.8 6l4.6 4.6L6 12Z"
    fill="#4D6475"
  />
</svg>
`;

exports[`Icon  ChevronRight matches snapshot 1`] = `
<svg
  fill="none"
  height="12"
  width="8"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M1.4 12 0 10.6 4.6 6 0 1.4 1.4 0l6 6-6 6Z"
    fill="#4D6475"
  />
</svg>
`;

exports[`Icon  CloseIcon matches snapshot 1`] = `
<svg
  fill="none"
  height="14"
  width="14"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M1.4 14 0 12.6 5.6 7 0 1.4 1.4 0 7 5.6 12.6 0 14 1.4 8.4 7l5.6 5.6-1.4 1.4L7 8.4 1.4 14Z"
    fill="#595D7B"
  />
  <path
    d="M1.4 14 0 12.6 5.6 7 0 1.4 1.4 0 7 5.6 12.6 0 14 1.4 8.4 7l5.6 5.6-1.4 1.4L7 8.4 1.4 14Z"
    fill="#000"
    fill-opacity="0.2"
  />
</svg>
`;

exports[`Icon  DonationWidgetLogo matches snapshot 1`] = `
<svg
  fill="none"
  height="56"
  width="58"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M54.125 52.125v-23.25 23.25ZM50 17.625c1.15 0 2.125-.4 2.925-1.2.8-.8 1.2-1.775 1.2-2.925s-.4-2.125-1.2-2.925c-.8-.8-1.775-1.2-2.925-1.2s-2.125.4-2.925 1.2c-.8.8-1.2 1.775-1.2 2.925s.4 2.125 1.2 2.925c.8.8 1.775 1.2 2.925 1.2ZM50 21c-2.15 0-3.938-.712-5.362-2.137C43.213 17.438 42.5 15.65 42.5 13.5c0-2.1.712-3.875 2.138-5.325C46.063 6.725 47.85 6 50 6c2.1 0 3.875.725 5.325 2.175C56.775 9.625 57.5 11.4 57.5 13.5c0 2.15-.725 3.938-2.175 5.363C53.875 20.288 52.1 21 50 21Zm-21-6.375c1.55 0 2.875-.537 3.975-1.613 1.1-1.075 1.65-2.412 1.65-4.012 0-1.55-.55-2.875-1.65-3.975-1.1-1.1-2.425-1.65-3.975-1.65-1.6 0-2.938.55-4.012 1.65-1.075 1.1-1.613 2.425-1.613 3.975 0 1.6.538 2.938 1.613 4.012 1.075 1.075 2.412 1.613 4.012 1.613ZM29 18c-2.55 0-4.688-.863-6.413-2.588C20.863 13.688 20 11.55 20 9c0-2.5.863-4.625 2.587-6.375C24.313.875 26.45 0 29 0c2.5 0 4.625.875 6.375 2.625S38 6.5 38 9c0 2.55-.875 4.688-2.625 6.412C33.625 17.137 31.5 18 29 18ZM15.125 31.125c0 1.8.887 3.862 2.663 6.188C19.562 39.638 23.125 43.2 28.474 48l.525.45.6-.525c5.15-4.7 8.65-8.2 10.5-10.5 1.85-2.3 2.775-4.4 2.775-6.3 0-1.7-.55-3.125-1.65-4.275-1.1-1.15-2.475-1.725-4.125-1.725-.95 0-1.85.225-2.7.675a9.682 9.682 0 0 0-2.325 1.725l-3.225 3.15h.3l-3.3-3.15c-.7-.7-1.463-1.275-2.288-1.725a5.474 5.474 0 0 0-2.662-.675c-1.65 0-3.025.575-4.125 1.725-1.1 1.15-1.65 2.575-1.65 4.275Zm-3.45 0c0-2.45.938-4.637 2.813-6.563 1.875-1.925 4.012-2.887 6.412-2.887 1.45 0 2.863.35 4.238 1.05a10.85 10.85 0 0 1 3.487 2.775l.375.45.375-.45c.95-1.15 2.1-2.075 3.45-2.775a9.16 9.16 0 0 1 4.275-1.05c2.35 0 4.475.962 6.375 2.887 1.9 1.925 2.85 4.113 2.85 6.563 0 2.5-.938 5-2.813 7.5-1.874 2.5-5.637 6.325-11.287 11.475L29 53.1l-3.15-2.925c-5.8-5.25-9.613-9.112-11.438-11.587-1.825-2.476-2.737-4.963-2.737-7.463ZM8 17.625c1.15 0 2.125-.4 2.925-1.2.8-.8 1.2-1.775 1.2-2.925s-.4-2.125-1.2-2.925c-.8-.8-1.775-1.2-2.925-1.2s-2.125.4-2.925 1.2c-.8.8-1.2 1.775-1.2 2.925s.4 2.125 1.2 2.925c.8.8 1.775 1.2 2.925 1.2ZM29.225 55.5v-3.375h24.9V29.85c0-.25-.1-.475-.3-.675-.2-.2-.425-.3-.675-.3h-9.075V25.5h9.075c1.2 0 2.225.425 3.075 1.275.85.85 1.275 1.875 1.275 3.075V55.5H29.225Zm-25.35-3.375h25.35V55.5H.5V29.85c0-1.2.425-2.225 1.275-3.075.85-.85 1.875-1.275 3.075-1.275h9.075v3.375H4.85c-.3 0-.537.1-.712.3a.999.999 0 0 0-.263.675v22.275ZM8 21c-2.15 0-3.938-.712-5.362-2.137C1.213 17.438.5 15.65.5 13.5c0-2.1.712-3.875 2.138-5.325C4.063 6.725 5.85 6 8 6c2.1 0 3.875.725 5.325 2.175C14.775 9.625 15.5 11.4 15.5 13.5c0 2.15-.725 3.938-2.175 5.363C11.875 20.288 10.1 21 8 21Z"
    fill="#423C66"
  />
</svg>
`;

exports[`Icon  Logo matches snapshot 1`] = `
<svg
  fill="none"
  height="32"
  width="147"
  xmlns="http://www.w3.org/2000/svg"
>
  <mask
    height="32"
    id="a"
    maskUnits="userSpaceOnUse"
    style="mask-type: alpha;"
    width="32"
    x="0"
    y="0"
  >
    <path
      d="M0 0h32v32H0z"
      fill="#D9D9D9"
    />
  </mask>
  <g
    mask="url(#a)"
  >
    <path
      d="M16.4 30.2a16.46 16.46 0 0 1-5.45-1.467 14.95 14.95 0 0 1-4.567-3.283c-1.322-1.389-2.394-3.044-3.217-4.967-.822-1.922-1.277-4.094-1.366-6.516 2.244 0 4.244.316 6 .95 1.755.633 3.25 1.61 4.483 2.933 1.234 1.322 2.195 3.006 2.883 5.05.69 2.045 1.1 4.478 1.234 7.3Zm-.433-12.433c-.712-1.156-1.467-2.095-2.267-2.817-.8-.722-1.822-1.383-3.067-1.983.111-.89.322-1.845.634-2.867.31-1.022.7-2.044 1.166-3.067.467-1.022 1-2.033 1.6-3.033.6-1 1.234-1.944 1.9-2.833a28.248 28.248 0 0 1 3.617 5.917 25.74 25.74 0 0 1 1.183 3.116 17 17 0 0 1 .633 2.933c-1 .312-1.972.89-2.916 1.734a14.519 14.519 0 0 0-2.483 2.9Zm3.4 11.633a50.335 50.335 0 0 0-.584-4.933c-.255-1.467-.583-2.678-.983-3.634 1.066-2.2 2.816-3.977 5.25-5.333 2.433-1.355 4.816-1.889 7.15-1.6-.356 4.178-1.45 7.533-3.284 10.067-1.833 2.533-4.35 4.344-7.55 5.433Z"
      fill="#423C66"
    />
  </g>
  <path
    d="M40 28.313h2.813V17.166h1.229l-3.01-1.907 12.364 13.053h2.594V13.28h-2.803v11.115h-1.24l3.022 1.896-12.375-13.01H40v15.03ZM62.594 28.51a7.54 7.54 0 0 0 1.99-.25 5.636 5.636 0 0 0 1.614-.729 3.62 3.62 0 0 0 1.083-1.125h.052v1.907h2.771v-7.01c0-.883-.236-1.64-.708-2.272-.472-.632-1.153-1.118-2.042-1.458-.889-.34-1.958-.51-3.208-.51-1.146 0-2.14.16-2.98.479-.833.312-1.5.746-2 1.302a3.605 3.605 0 0 0-.906 1.885l-.02.125h2.635l.031-.094c.14-.507.48-.91 1.021-1.208.549-.299 1.274-.448 2.177-.448 1.035 0 1.83.202 2.386.604.562.403.843.97.843 1.698v2.417c0 .555-.17 1.042-.51 1.458-.333.41-.806.73-1.417.959-.604.222-1.312.333-2.125.333-.84 0-1.517-.132-2.031-.396-.507-.27-.76-.677-.76-1.219v-.02c0-.424.18-.76.541-1.01.368-.25.959-.386 1.771-.407l5.927-.146v-1.656l-6.312.125c-1.098.027-1.993.177-2.688.448-.687.27-1.19.642-1.51 1.114-.32.473-.48 1.025-.48 1.657v.01c0 .743.199 1.371.594 1.885.403.507.97.893 1.698 1.157.73.264 1.584.395 2.563.395ZM78.083 28.5c.361 0 .705-.014 1.032-.042.333-.027.632-.066.895-.114v-1.98c-.16.028-.354.053-.583.073a7.15 7.15 0 0 1-.667.032c-.812 0-1.399-.146-1.76-.438-.354-.298-.531-.809-.531-1.531v-5.167h3.541v-2.02H76.47v-2.834h-2.781v2.834h-2.334v2.02h2.334v5.47c0 1.277.354 2.214 1.062 2.812.715.59 1.826.885 3.333.885ZM86.792 28.563c.722 0 1.382-.098 1.979-.292a4.949 4.949 0 0 0 1.573-.844 4.16 4.16 0 0 0 1.073-1.27h.062v2.155h2.771v-11h-2.77v5.47c0 .555-.095 1.055-.282 1.5-.18.444-.438.822-.77 1.135a3.442 3.442 0 0 1-1.178.718c-.451.16-.951.24-1.5.24-1.049 0-1.844-.288-2.385-.865-.542-.576-.813-1.395-.813-2.458v-5.74h-2.77v6.25c0 1.056.2 1.959.603 2.709.41.743.986 1.312 1.73 1.708.75.39 1.642.584 2.677.584ZM96.417 28.313h2.77v-5.459c0-.743.14-1.368.417-1.875a2.69 2.69 0 0 1 1.188-1.156c.514-.264 1.118-.396 1.812-.396.243 0 .476.01.698.031.222.021.42.049.594.084v-2.344a5.301 5.301 0 0 0-1.063-.104c-.902 0-1.666.215-2.291.646-.625.43-1.06 1.027-1.302 1.791h-.052v-2.208l-2.771-.01v11ZM105.469 28.469c.451 0 .826-.146 1.125-.438.305-.298.458-.66.458-1.083 0-.43-.153-.792-.458-1.083-.299-.299-.674-.448-1.125-.448-.452 0-.83.149-1.136.448a1.456 1.456 0 0 0-.448 1.083c0 .424.15.785.448 1.083.306.292.684.438 1.136.438ZM113.573 28.51a7.55 7.55 0 0 0 1.99-.25 5.63 5.63 0 0 0 1.614-.729c.465-.32.826-.694 1.083-1.125h.053v1.907h2.77v-7.01c0-.883-.236-1.64-.708-2.272-.472-.632-1.153-1.118-2.042-1.458-.889-.34-1.958-.51-3.208-.51-1.146 0-2.139.16-2.979.479-.833.312-1.5.746-2 1.302a3.604 3.604 0 0 0-.906 1.885l-.021.125h2.635l.031-.094c.139-.507.48-.91 1.021-1.208.549-.299 1.275-.448 2.177-.448 1.035 0 1.83.202 2.386.604.562.403.844.97.844 1.698v2.417c0 .555-.171 1.042-.511 1.458-.333.41-.805.73-1.417.959-.604.222-1.312.333-2.125.333-.84 0-1.517-.132-2.031-.396-.507-.27-.76-.677-.76-1.219v-.02c0-.424.18-.76.541-1.01.368-.25.959-.386 1.771-.407l5.927-.146v-1.656l-6.312.125c-1.097.027-1.993.177-2.688.448-.687.27-1.191.642-1.51 1.114-.32.473-.479 1.025-.479 1.657v.01c0 .743.198 1.371.594 1.885.402.507.968.893 1.697 1.157.73.264 1.584.395 2.563.395ZM123.313 28.313h2.77V13.28h-2.77v15.031ZM128.417 28.313h2.771V13.28h-2.771v15.031ZM135.583 32c.91 0 1.695-.125 2.355-.375a4.38 4.38 0 0 0 1.75-1.208c.506-.556.993-1.302 1.458-2.24l5.229-10.864h-2.917l-4.052 9.062.532-.458h-.73l.532.458-4.261-9.063h-2.958l5.448 10.928-.271.479c-.278.451-.618.764-1.021.937-.403.18-.892.271-1.469.271a6.03 6.03 0 0 1-1.291-.146v2.052c.222.049.475.087.76.115.292.035.594.052.906.052Z"
    fill="#423C66"
  />
</svg>
`;
